lib/ansible/module_utils/known_hosts.py

```diff
@@ -190,7 +190,10 @@ def add_host_key(module, fqdn, key_type="rsa", create_dir=False, port=SSH_KEYSCA
     elif not os.path.isdir(user_ssh_dir):
         module.fail_json(msg="%s is not a directory" % user_ssh_dir)
 
-    this_cmd = "%s -t %s %s" % (keyscan_cmd, key_type, fqdn)
```

Comment:
I think simply doing this would be better:
<pre>
if port is None:
    port = SSH_KEYSCAN_DEFAULT_PORT
this_cmd = "%s -t %s %s -p %s" % (keyscan_cmd, key_type, fqdn, port)
</pre>
There's no reason to conditionally exclude the `-p` option, it should always be set.